File: 2 term/PB/PB Lab2.5/PB Lab2.5 py/Prism3.py
import math

from Prism import Prism


class Prism3(Prism):
    def calculate_surface(self):
        return 2 * self.calculate_base_surface() + 3 * self.height * self.side_length

    def calculate_base_surface(self):
        return self.side_length ** 2 * math.sqrt(3) / 4

File: 2 term/PB/PB Lab2.5/PB Lab2.5 cs/Prism3.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PB_Lab2._5_cs
{
    public class Prism3 : Prism
    {
        public Prism3(double height, double sideLength) : base(height, sideLength)
        {}

        public override double CalculateBaseSurface()
        {
            return SideLength * SideLength * Math.Sqrt(3) / 4;
        }

        public override double CalculateSurface()
        {
            return CalculateBaseSurface() * 2 + SideLength * Height * 3;
        }
    }
}
